apollo-federation/src/supergraph/mod.rs

```diff
@@ -1693,103 +1653,6 @@ lazy_static! {
     };
 }
 
-fn insert_directive(
-    schema: &mut FederationSchema,
-    pos: &CompositeTypeDefinitionPosition,
-    directive: Component<Directive>,
-) -> Result<(), FederationError> {
-    match pos {
-        CompositeTypeDefinitionPosition::Union(pos) => pos.insert_directive(schema, directive),
-        CompositeTypeDefinitionPosition::Object(pos) => pos.insert_directive(schema, directive),
-        CompositeTypeDefinitionPosition::Interface(pos) => pos.insert_directive(schema, directive),
-    }
-}
-
-trait CompositeType {
-    fn directives(&self) -> &DirectiveList;
-}
-
-impl CompositeType for UnionType {
-    fn directives(&self) -> &DirectiveList {
-        &self.directives
-    }
-}
-
-impl CompositeType for ObjectType {
-    fn directives(&self) -> &DirectiveList {
-        &self.directives
-    }
-}
-
-impl CompositeType for InterfaceType {
-    fn directives(&self) -> &DirectiveList {
-        &self.directives
-    }
-}
-
-fn apply_context_to_type<T>(
-    ty: &Node<T>,
-    subgraphs: &mut FederationSubgraphs,
-    graph_enum_value_name_to_subgraph_name: &IndexMap<Name, Arc<str>>,
-    federation_spec_definitions: &IndexMap<Name, &'static FederationSpecDefinition>,
-    context_name_in_supergraph: &Name,
-    pos: &CompositeTypeDefinitionPosition,
-) -> Result<(), FederationError>
-where
-    T: CompositeType,
-{
-    for directive in ty.directives().get_all(context_name_in_supergraph.as_str()) {
-        FederationSpecDefinition::context_directive_arguments(directive).and_then(
-            |context_name| {
-                let mut arr = context_name.name.split("__");
```

Comment:
Bugs:
- This should be an `rsplit` instead of `split`, since context names can't have underscores but subgraph names can.